### PR TITLE
Signature help option and keyboard shortcuts

### PR DESCRIPTION
**Pull request by microbit-matt-hillsdon**
Support the same keyboard interactions as VS Code:
- Mod-Shift-Space to open
- Escape to close

Add an automatic/manual option in the settings.

We had a brief discussion about naming and agreed to go with "Parameter help" for now. VS Code uses "Parameter hints". LSP uses "Signature help" but that's a more complicated notion for our users. We might revisit.

### CODE REVIEW

File: lang/fr.json [hunk 1]
```diff
@@ -374,6 +374,18 @@
...
+    "description": "Setting label to control whether pop-up documentation for function/method parameters is automatically shown."
+  },
+  "parameter-hints-automatic": {
+    "defaultMessage": "Parameter hints setting for when the parameter documentation is shown automatically",
+    "description": "Automatic"
+  },
+  "parameter-hints-manual": {
+    "defaultMessage": "Parameter hints setting for when the user must press a key combination to open the parameter documentation",
+    "description": "Manual"
+  },
```
Comment (microbit-matt-hillsdon) [inline lang/fr.json:388]: We'll live with this in English until retranslation for beta.

**Comment by microbit-matt-hillsdon** [comment]
Code change looks good now, apart from the weird naming inconsistency that we should discuss.

**Comment by microbit-matt-hillsdon** [comment]
> Code change looks good now, apart from the weird naming inconsistency that we should discuss.

We discussed briefly and went with "Parameter help". Might revisit.

File: src/settings/SettingsArea.tsx [hunk 2]
```diff
@@ -44,7 +50,24 @@ const SettingsArea = () => {
...
+        codeStructureOptions,
+        "highlight-code-structure",
+        intl
+      ),
+      parameterHelp: createOptions(
+        parameterHelpOptions,
+        "parameter-help",
+        intl,
+        {
+          shortcut: (isMac ? "Cmd" : "Ctrl") + "-Shift+Space",
```
Comment (microbit-robert) [inline src/settings/SettingsArea.tsx:66]:
```suggestion
          shortcut: (isMac ? "Cmd" : "Ctrl") + "+Shift+Space",
```

Comment (microbit-matt-hillsdon) [reply src/settings/SettingsArea.tsx:66]: Fair enough, I think my punctuation is less common (though it's not like you can press them in any order).

**Comment by microbit-matt-hillsdon** [comment]
[e4fd137](https://github.com/microbit-foundation/python-editor-next/pull/631/commits/e4fd13746fa3c54adefc555851da0d66573e0eee) updates the script to remove keys not in `en.json`.

**Comment by microbit-robert** [comment]
> [e4fd137](https://github.com/microbit-foundation/python-editor-next/pull/631/commits/e4fd13746fa3c54adefc555851da0d66573e0eee) updates the script to remove keys not in `en.json`.

Looks good!

**Pull request merged**